rats-tls/src/attesters/sev/ttrpc/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+
+mod protocols;
+mod sev_evidence;
+
+#[macro_use]
+extern crate log;
+
+const SERVER_ADDR: &str = "vsock:///tmp/aeb.sock";
+const SERVER_PORT: &str = "5577";
+
+#[no_mangle]
+pub extern "C" fn retrieve_attestation_evidence(
+    guest_handle: u32,
+    evidence_size: u32,
+) -> Option<Box<SevEvidence>> {
+    let req = aeb::RetrieveAttestationEvidenceRequest {
+        guest_handle: guest_handle,
+        evidence_size: evidence_size,
+        unknown_fields: Default::default(),
+        cached_size: Default::default(),
+    };
+
+    let agent_client = connect().context("connect");
+    match agent_client {
+        Ok(agent_client) => match agent_client.retrieve_attestation_evidence(default_ctx(), &req) {
+            Ok(res) => {
+                let evidence = res.get_evidence();
+                let wrap_command = String::from_utf8(evidence.to_vec())
+                    .and_then(|evidence| Ok(serde_json::from_str::<SevEvidence>(&evidence[..])))
+                    .unwrap();
+
+                match wrap_command {
+                    Ok(wrap_command) => {
+                        debug!("sev evidence is {:#X?}", wrap_command);
+                        return Some(Box::new(wrap_command));
+                    }
+                    Err(e) => {
+                        error!("failed to deserialize aeb respense to sev evidence {:?}", e);
+                        return None;
+                    }
+                }
+            }
+            Err(e) => {
+                error!("failed to retrieve evidence from aeb service {:?}", e);
+                return None;
+            }
+        },
+        Err(e) => {
+            error!("failed to create ttrpc client {:?}", e);
+            None
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn retrieve_attestation_evidence_size(guest_handle: u32) -> u32 {
+    let req = aeb::RetrieveAttestationEvidenceSizeRequest {
+        guest_handle: guest_handle,
+        unknown_fields: Default::default(),
+        cached_size: Default::default(),
+    };
+
+    let agent_client = connect().context("connect");
+
+    match agent_client {
+        Ok(agent_client) => {
+            match agent_client.retrieve_attestation_evidence_size(default_ctx(), &req) {
+                Ok(res) => res.evidence_size,
+                Err(e) => {
+                    error!("failed to retrieve evidence size from aeb service {:?}", e);
+                    0
+                }
+            }
+        }
+        Err(e) => {
+            error!("failed to create ttrpc client {}", e);
+            0
+        }
+    }
+}
+
+pub fn connect() -> Result<aeb_ttrpc::AebClient> {
+    let addr = format!("{}:{}", SERVER_ADDR, SERVER_PORT);
+    debug!("connect {}", addr);
+
+    let c = Client::connect(&addr[..]).context(format!("failed to connect {}", addr))?;
+
+    Ok(aeb_ttrpc::AebClient::new(c))
+}
+
+fn default_ctx() -> ttrpc::context::Context {
+    let ctx = ttrpc::context::with_timeout(0);
+
+    ctx
+}
```

Comment:
I think the value can be returned directly, like
fn default_ctx() -> ttrpc::context::Context {
    ttrpc::context::with_timeout(0)
}